File: Code/BeforeLegends/Assets/Scripts/World Map/Controls/MoveOnClick.js
//#pragma strict

var objData : MapObjectCarrier;
var mouseOverUIElement : boolean;

//function Update(){
//    if(Input.GetMouseButtonDown(0) && !mouseOverUIElement){
//		if(objData == null){
//			objData = gameObject.GetComponent(MapObjectCarrier);
//			if(objData == null){
//				objData = gameObject.AddComponent(MapObjectCarrier);
//				objData.setPosition(Vec2i(MouseTileInput.instance.lastTile.x, MouseTileInput.instance.lastTile.y));
//				transform.position = WorldMapData.getInstance().tiles[objData.pos.x, objData.pos.y].position; 
//				InterfaceData.getInstance().selectedCarrier = objData;
//				FogOfWar.instance.CheckTiles(Vec2i(objData.pos.x, objData.pos.y), FogOfWar.instance.visionRange);
//				FogOfWar.instance.SetEntitiesToVisible();
//				return;
//			}
//		}
//		var path : Vec2i[] = WorldMapData.getInstance().findPath(objData.pos.x, objData.pos.y, MouseTileInput.instance.lastTile.x, MouseTileInput.instance.lastTile.y, objData.movedMax - objData.moved, false);
//		objData.followPath(path, 0.25);
//	}
//}

//function SetMouseOverUIElement(yesno : boolean)
//{
//    mouseOverUIElement = yesno;
//}

File: Code/BeforeLegends/Assets/Scripts/World Map/Controls/MoveOnClick.cs
using UnityEngine;
using System.Collections;

public class MoveOnClick : MonoBehaviour {

    public MapObjectCarrier objData;
    public bool mouseOverUIElement;

    void Update(){
        if(Input.GetMouseButtonDown(0) && !mouseOverUIElement)
        {
            //if (UnityEngine.EventSystems.EventSystem.current.IsPointerOverGameObject())
                //return;
		    if(objData == null)
            {
			    objData = gameObject.GetComponent<MapObjectCarrier>();
			    if(objData == null)
                {
				    objData = gameObject.AddComponent<MapObjectCarrier>();
				    objData.setPosition(new Vec2int(MouseTileInput.instance.lastTile.x, MouseTileInput.instance.lastTile.y));
				    transform.position = WorldMapData.instance.tiles[objData.pos.x, objData.pos.y].position; 
				    InterfaceData.instance.selectedCarrier = objData;
				    FogOfWar.instance.CheckTiles(new Vec2int(objData.pos.x, objData.pos.y), FogOfWar.instance.visionRange);
				    FogOfWar.instance.SetEntitiesToVisible();
				    return;
			    }
		    }

		    Vec2int[] path = WorldMapData.instance.findPath(objData.pos.x, objData.pos.y, MouseTileInput.instance.lastTile.x, MouseTileInput.instance.lastTile.y, objData.movedMax - objData.moved, false);
            objData.followPath(path, 0.25f);
	    }
    }

    public void SetMouseOverUIElement(bool yesno)
    {
        mouseOverUIElement = yesno;
    }
}
